service-level-indicators/spark-retrieval-success-rate.md

```diff
@@ -0,0 +1,612 @@
+4. Retrieve the root block of the content identified by `CID` from that multiaddr using Graphsync or the [IPFS Trustless Gateway protocol](https://specs.ipfs.tech/http-gateways/trustless-gateway/) (over HTTP). It uses the protocol advertised to IPNI for the retrieval. If both protocols are advertised, then it chooses HTTP.
+5. Verify that the received block matches the `CID`.
+
+## Reporting Measurements to Spark-API
+
+When the Spark Checkers have completed a retrieval and determined whether or not the CID is retrievable by following the above steps, they report their result, which we call a measurement to the Spark-API service. Spark-API ingests these measurements, batches them into chunks of up to 100k measurements, uploads each chunk to [web3.storage](http://web.storage) (now Storacha), and commits the batch CID to the Spark [smart contract](https://github.com/filecoin-station/spark-impact-evaluator). We call this the [Spark-Publish logic](https://github.com/filecoin-station/spark-api/tree/main/publish).
+
+<aside>
+ℹ️
+
+Failed retrieval attempts are also reported with a reason code for the failure reason.  Failure reasons and how they map to RSR are covered in [Retrieval Result Mapping to RSR](#retrieval-result-mapping-to-rsr).
+
+</aside>
+
+## Evaluating Measurements with Spark-Evaluate
+
+[Spark-Evaluate](https://github.com/filecoin-station/spark-evaluate) is the Spark service that evaluates each measurement to decide whether or not it is valid. It listens out for on-chain events that indicate that the Spark Publish logic has posted a commitment on chain. It then takes the CID of the on chain commitment and fetches the corresponding measurements from Storacha.
+
+Once Spark-Evaluate retrieves the measurements, it does “fraud detection” to remove all unwanted [Retrieval Task Measurement](#retrieval-task-measurement)s as summarized below:
+
+| [Retrieval Task Measurement](#retrieval-task-measurement)s which are removed | Why removed? |
+| --- | --- |
+| Those for [Eligible Deal](#eligible-deal)s not in the round | To prevent checkers from checking any [Eligible Deal](#eligible-deal) of their choosing either to inflate or tarnish an SP’s stats. |
+| Those which are for [**Retrieval Task**](#retrieval-task)s that are not within the $k$-closest for a checker | Same principle as above.  It’s not good enough to pick any [**Retrieval Task**](#retrieval-task) from the [Round Retrieval Task List](#round-retrieval-task-list). |
+| Those which are submitted after the first $k$ measurements from a given IPV4 /24 subnet.
+(i.e, The first $k$ [Retrieval Task Measurement](#retrieval-task-measurement)  within a IPv4 /24 subnet are accepted.  Others are rejected.) | Prevent a malicious actor from creating tons of station ids that are then used to “stuff the ballot box” from one node. IPV4 /24 subnets are being used here as a scarce resource. |
```

Comment:
```suggestion
| Those which are for [**Retrieval Task**](#retrieval-task)s that are not within the $k$-closest for a checker | Same principle as above.  It’s not good enough to pick any [**Retrieval Task**](#retrieval-task) from the [Round Retrieval Task List](#round-retrieval-task-list). |
| Those which are submitted after the first $k$ measurements from a given IPV4 /24 subnet. (i.e, The first $k$ [Retrieval Task Measurement](#retrieval-task-measurement)  within a IPv4 /24 subnet are accepted.  Others are rejected.) | Prevent a malicious actor from creating tons of station ids that are then used to “stuff the ballot box” from one node. IPV4 /24 subnets are being used here as a scarce resource. |
```

To fix table rendering (there must be no line breaks inside cells):

<img width="962" alt="Screenshot 2024-11-04 at 17 21 28" src="https://github.com/user-attachments/assets/dddb40d7-5309-4bdc-b31f-b9a7b0461eb7">